modules/volunteers/admin/sitesupport/checkout.js

```diff
@@ -0,0 +1,223 @@
+/* jshint esversion: 6 */
+/* globals  apiRequest, showSidebar, hideSidebar, confirmbox, alertbox, showSpinner, hideSpinner */
+
+export default {
+  data() {
+    return {
+      checkout: [],
+      hoursSpent: 0,
+      groupsNow: [],
+      failedList: null,
+    }
+  },
+  methods: {
```

Comment:
It might make the code a bit easier to read if we extract some of these functions to another spot in the file, and then reference those functions in our `methods` object.

For example, you might have something like:

```
const processCheckout = () => {
  // Code to complete the checkout process
}

export default {
  data() {
    return {
       checkout: [],
       hoursSpent: 0,
       groupsNow: [],
       failedList: undefined // recommend undefined here instead of null, as null can have weird behaviors in JS and undefined is more typical unless Vue specifically requires null
    },
    methods: {
       processCheckout
    }
  }
}
```

Reply:
Actually, on seeing `failedList` being used below I'd recommend simply setting to `[]` instead as both the default and for cases where no items have failed to be checked out.

Reply:
Makes sense I will look and dig.